### PR TITLE
Landing update

### PR DESCRIPTION
**Pull request by Exilliar**
Quite a few little updates to the landing page on this pr. The main ones are:

- Small refactor of the file structure. Moved each component into a file named by the view that the component is used in, with a shared folder for components that are/can be used in multiple views. There is also a "core" folder for components that are used by `App.vue`
- Split up the landing page into more components to keep the file a little shorter
- Updated the Building Element Category chips to be in their own component in the `shared` folder so that they can be used on other pages and have the correct colours
- Added a loading spinner to the landing page while the reports are being loaded
- Added some error handling so that if the loading of the reports fails then a "retry" button is shown to the user
- Fixed the darkmode problems that were on the page
- Made the searchbar work
- The graphs use the colour from the materials, rather than using their own defaults
- Added some comments here and there
- Fixed some bugs
- Added in the ability to delete reports. To test this, create a report, then use the settings button on the project card (the cog wheel), then click "delete". A confirmation modal should pop up, and if it is successful a snackbar will appear saying that and the report should be removed from the list (the reports are re-gathered from Speckle to ensure that the list is up to date)
- removed some unneccassary code

Due to some of the refactor stuff there may be some conflicts created on other branches, so it'll probably be a good idea for everyone to do a `git pull origin dev` after this pr has been merged.

Also, I'm hoping for this to be the last major pr needed on the landing page for a little bit, so please be as nitpicky as possible. Anything that you think should be in the v1 release can be commented on

### CODE REVIEW

File: src/components/shared/BECChip.vue [hunk 1]
```diff
@@ -0,0 +1,70 @@
+<template>
+  <v-chip :color="backgroundColor" :style="`color: ${color}`">{{ category }}</v-chip>
+</template>
+<script lang="ts">
+import { Vue, Component, Prop } from "vue-property-decorator";
+
+type BECCategory =
```
Comment (tomhebunn) [inline src/components/shared/BECChip.vue:7]: can we pull this from store? keen not to duplicate lists. suggest:

`getBECs() {
    return $store.state.buildingElementCategories
}`

File: src/components/shared/BECChip.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  @Prop() category!: BECCategory;
+
+  get backgroundColor() {
+    return this.becs.find((b) => b.name === this.category)?.backgroundColor;
+  }
+  get color() {
+    return this.becs.find((b) => b.name === this.category)?.color;
+  }
+
+  becs: BEC[] = [
```
Comment (tomhebunn) [inline src/components/shared/BECChip.vue:32]: can we store these as objects within store as an updated buildingElementCategories list?

File: src/components/shared/BECChip.vue (hunk 1 above)
Comment (tomhebunn) [reply src/components/shared/BECChip.vue:7]: See below. Worth updating the store buildingElementCategories list as a list of objects and pull through all key values for name

**Review by Kchurch2 — COMMENTED** [review]
Hi Rory,

looks good, just one typo on the delete message

File: src/store/index.ts [hunk 3]
```diff
@@ -57,21 +61,46 @@ export default new Vuex.Store({
...
-      "Superstructure",
-      "Mechanical Services",
-      "Electrical Services",
-      "Public Health & Hydraulics",
-      "Building Envelope",
-      "Space Plan",
+    availableRegions: ["India", "UK"],
+    becs: [
+      {
+        name: "Superstructure" as BECCategory,
```
Comment (tomhebunn) [inline src/store/index.ts:70]: BECCategory should be changed to "BECategory". BEC = Building Element Category. BECCategory = Building Element Category Category

**Review by tomhebunn — APPROVED** [review]

**Comment by tomhebunn** [comment]
One minor comment that BECCategory would be "Building Element Category Category". Happy to merge once fixed.

**Pull request merged**